### PR TITLE
Small KeytoolCertificateGenerator improvements 

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Refactoring

### Description

* remove several misleading references to `JKS` in the implementation.  The implementation always emitted `PKCS12` format key-stores - the references to `JKS` were actually ineffective.
* announce the keystore type in the kafka broker/client config.  Kafka assumes `JKS`. We were feeding it `PKCS12` which it was somehow accepting.  This change brings consistency.
* eliminate the use of `openssl`.  The implementation still shells-out to keytool, but at least this is one less dependency.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KeytoolCertificateGenerator.java [hunk 1]
```diff
@@ -137,10 +139,8 @@ public void generateTrustStore(String certFilePath, String alias) throws General
     public void generateTrustStore(String certFilePath, String alias, String trustStoreFilePath)
             throws GeneralSecurityException, IOException {
         // keytool -import -trustcacerts -keystore truststore.jks -storepass password -noprompt -alias localhost -file cert.crt
-        KeyStore keyStore = KeyStore.getInstance("JKS");
         if (Path.of(trustStoreFilePath).toFile().exists()) {
-            keyStore.load(new FileInputStream(trustStoreFilePath), getPassword().toCharArray());
-
+            var keyStore = KeyStore.getInstance(new File(trustStoreFilePath), getPassword().toCharArray());
```
[the remark this answers is not in the document]
Comment (k-wall) [reply impl/src/main/java/io/kroxylicious/testing/kafka/common/KeytoolCertificateGenerator.java:146]: Added annotation to the class explaining the use-case.

**Review by franvila — APPROVED** [review]

**Review by SamBarker — APPROVED** [review]
It fixes the dependency on OpenSSL for me.

Thanks @k-wall

File: impl/src/test/java/io/kroxylicious/testing/kafka/common/KeytoolCertificateGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.List;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class KeytoolCertificateGeneratorTest {
+    @Test
+    public void generatesKeyStore() throws Exception {
```
[the remark this answers is not in the document]
Comment (k-wall) [reply impl/src/test/java/io/kroxylicious/testing/kafka/common/KeytoolCertificateGeneratorTest.java:22]: My motivation for adding these tests was to do the bare minimum to cover the areas of change.  I don't think 
`KeytoolCertificateGenerator` is really what we want (see #131), so I don't want to put effort polishing something that we may throw away.

The rest of the Junit5 extension code only uses the no-arg KeytoolCertificateGenerator() constructor, so I wanted the test to do the same thing.   That gives KeytoolCertificateGenerator the responsibility to managed the lifecycle of the test-dir.
